woocommerce/woocommerce-gutenberg-products-block
# Move `CheckboxControl` to components package and leave alias in checkout package

## What

Move `CheckboxControl` to components package and leave alias in checkout package.

## Why

Doesn't make sense to have external components in multiple locations in the codebase. An alias is left in the checkout package for backward compatibility.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Install [WooCommerce Stripe Payment Gateway](https://wordpress.org/plugins/woocommerce-gateway-stripe/) and set it up
2. Edit the Checkout page in the Site editor, select the "Terms and Conditions block" and set the "Require checkbox" option to true.
3. Ensure that order notes are enabled on the block. Ensure the checkboxes for these blocks all appear and the block as a whole renders OK.
4. As a logged-in user, add an item to your cart and go to the Checkout block in the front-end.
5. Ensure the "Use same address for billing" checkbox is available and works as intended.
6. Scroll down to the Terms and conditions, ensure the checkbox renders and works correctly.
7. Check the "Add a note to your order" box. Ensure it works and the textarea renders.
8. Check out using Stripe and ensure the "Save payment information to my account for future purchases." renders.
9. Check this box and check out, ensure checkout is successful.
10. Add an item to your cart again and go back to the Checkout block. Ensure your card was saved.
11. In the site editor, create a page and add the "All Products" and "Filter by Attribute" blocks.
12. On "Filter by Attribute" choose an attribute with multiple terms and ensure the checkboxes show up in the editor.
13. View this page on the front-end and ensure they render there, too.

### Internal Developer Testing (do not include in the notes)

1. Open a page with the WooCommerce Checkout block on.
2. Open the JS console and type `wc.blocksComponents`
3. Ensure CheckboxControl is there and it is not undefined.
4. Do the same for `wc.blocksCheckout`.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> We have moved the `CheckboxControl` component to the @woocommerce/blocks-components package. Previously this was available in @woocommerce/blocks-checkout . Your code will continue to work as we have added aliases to the new location. Despite this, it is recommended that you change your code to import this component from `@woocommerce/blocks-components` as the import from the checkout package will be deprecated in the future.
